Library/Homebrew/brew.sh

```diff
@@ -427,14 +436,16 @@ Your curl executable: $(type -p $HOMEBREW_CURL)"
 Minimum required version: ${HOMEBREW_MINIMUM_GIT_VERSION}
 Your Git version: $major.$minor.$micro.$build
 Your Git executable: $(unset git && type -p $HOMEBREW_GIT)"
-    if [[ -z $HOMEBREW_GIT_PATH || -z $HOMEBREW_DEVELOPER ]]; then
+
+    if [[ -n $HOMEBREW_GIT_PATH || -z $HOMEBREW_DEVELOPER ]]
+    then
+      odie "$message"
+    else
       HOMEBREW_FORCE_BREWED_GIT="1"
       if [[ -z $HOMEBREW_GIT_WARNING ]]; then
         onoe "$message"
         HOMEBREW_GIT_WARNING=1
       fi
-    else
-      odie "$message"
```

Comment:
I actually think the only situation where we should print a warning any more is if the `HOMEBREW_GIT_PATH` `git` is too old as:
- if the system `git` is too old: we will set `HOMEBREW_FORCE_BREWED_GIT ` and proceed accordingly
- if they have already set `HOMEBREW_FORCE_BREWED_GIT`: we don't want to print a warning/error but instead proceed to try and install it (in `update.sh` which may fail, at which point we should tell them about `HOMEBREW_GIT_PATH`)

Thoughts?